tests/unittests/halfTest.cpp

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright (c) 2018-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// This file tests the basic functionality of the half type.
+// This is by no mean a test to show the IEEE 754 compliance!
+
+#include "glow/Support/Half.h"
+#include "glow/Base/Type.h"
+
+#include "gtest/gtest.h"
+
+using namespace glow;
+
+TEST(Half, add) {
+  half a = 2.0;
+  half b = 1.0;
+  EXPECT_EQ(a + b, half(float(a) + float(b)));
+}
+
+TEST(Half, sub) {
+  half a = 2.0;
+  half b = 0.5;
+  EXPECT_EQ(a - b, half(float(a) - float(b)));
+}
+
+TEST(Half, mul) {
+  half a = 3.5;
+  half b = 3.0;
+  EXPECT_EQ(a * b, half(float(a) * float(b)));
+}
+
+TEST(Half, div) {
+  half a = 3.5;
+  half b = 3.0;
+  EXPECT_EQ(a / b, half(float(a) / float(b)));
+}
+
+TEST(Half, gt) {
+  half a = 3.5;
+  half b = 3.0;
+  EXPECT_EQ(a > b, float(a) > float(b));
+  EXPECT_TRUE(a > b);
+}
+
+TEST(Half, lt) {
+  half a = 3.5;
+  half b = 3.0;
+  EXPECT_EQ(a < b, float(a) < float(b));
+  EXPECT_FALSE(a < b);
+}
+
+TEST(Half, eq) {
+  half a = 3.5;
+  half b = 3.0;
+  EXPECT_EQ(a == b, float(a) == float(b));
```

Comment:
All of the inputs (a, b) in all of the tests are the same. We could just combine all of them into one long test instead of having multiple tests.

Reply:
I'll change the values. I personally like the fact that the tests have different "names".